src/main/java/dev/harrel/jsonschema/Applicators.java

```diff
@@ -305,20 +322,29 @@ class IfThenElseEvaluator implements Applicator {
     }
 
     @Override
-    public boolean apply(EvaluationContext ctx, JsonNode node) {
+    public Set<String> getVocabularies() {
+        return APPLICATOR_VOCABULARY;
+    }
+
+    @Override
+    public Result evaluate(EvaluationContext ctx, JsonNode node) {
         if (ctx.resolveInternalRefAndValidate(ifRef, node)) {
-            return thenRef
+            boolean valid = thenRef
                     .map(ref -> ctx.resolveInternalRefAndValidate(ref, node))
                     .orElse(true);
+
+            return valid ? Result.success() : Result.failure("Object matched against If but failed matching against Then schema");
```

Comment:
Suggesting `Value matches against schema from 'if' but does not match against schema from 'then'`